include/picongpu/fields/incidentField/profiles/GaussianPulse.hpp

```diff
@@ -136,29 +127,37 @@ namespace picongpu::fields::incidentField
                  */
                 HINLINE GaussianPulseFunctorIncidentE(float_X const currentStep, float3_64 const unitField)
                     : Base(currentStep, unitField)
+                    , laguerre_norm(getLaguerreNorm())
                 {
                     // This check is done here on HOST, since std::numeric_limits<float_X>::epsilon() does not
                     // compile on laserTransversal(), which is on DEVICE.
-                    auto etrans_norm = 0.0_X;
-                    for(uint32_t m = 0; m < Unitless::laguerreModes.size(); ++m)
-                        etrans_norm += Unitless::laguerreModes[m];
                     PMACC_VERIFY_MSG(
-                        math::abs(etrans_norm) > std::numeric_limits<float_X>::epsilon(),
+                        math::abs(laguerre_norm) > std::numeric_limits<float_X>::epsilon(),
                         "Sum of laguerreModes can not be 0.");
                 }
 
+                //!
+                HINLINE static float_X getLaguerreNorm()
+                {
+                    float_X value = 0.0_X;
+                    for(uint32_t m = 0; m < Unitless::laguerreModes.size(); ++m)
+                        value += Unitless::laguerreModes[m];
+
+                    return value;
```

Comment:
Should be
```suggestion
                    using namespace std;
                    return reduce(cbegin(Unitless::laguerreModes), cend(Unitless::laguerreModes), std::Plus<float_X>{});
```
but @psychocoderHPC explained that `pmacc::Vector` doesn't have an iterator interface.